### PR TITLE
CORE-512: Handle a BDB blockHeight of `-1` (not supported)

### PR DESCRIPTION
**Pull request by EBGToo**
Java changes pending.  Note: in the blockchain+currencies declarations 'ethereum-testnet' changed to 'ethereum-ropsten' to be consistent with BDB.

### CODE REVIEW

File: Swift/BRCrypto/BRCryptoSystem.swift [hunk 1]
```diff
@@ -76,30 +182,44 @@ public final class System {
...
-        default: return AddressScheme (core: CRYPTO_ADDRESS_SCHEME_GEN_DEFAULT)
-        }
+        return defaultAddressSchemeMap[network.uids] ?? .genDefault
     }
 
     ///
+    /// Wallet Manager Modes
+    ///
+
+    var supportedModesMap: [String:[WalletManagerMode]] = [
```
Comment (michael-brd) [inline Swift/BRCrypto/BRCryptoSystem.swift]: This should really be at the C level, but time is a factor.

File: Swift/BRCrypto/BRCryptoSystem.swift [hunk 2]
```diff
@@ -76,30 +182,44 @@ public final class System {
...
+        "bitcoin-mainnet":      [.p2p_only],
```
Comment (michael-brd) [inline Swift/BRCrypto/BRCryptoSystem.swift]: Are we missing api_only for bitcoin?

File: Swift/BRCrypto/BRCryptoSystem.swift [hunk 3]
```diff
@@ -76,30 +182,44 @@ public final class System {
...
-        case Currency.codeAsBTC: return AddressScheme (core: CRYPTO_ADDRESS_SCHEME_BTC_SEGWIT)
-        case Currency.codeAsBCH: return AddressScheme (core: CRYPTO_ADDRESS_SCHEME_BTC_LEGACY)
-        case Currency.codeAsETH: return AddressScheme (core: CRYPTO_ADDRESS_SCHEME_ETH_DEFAULT)
-        default: return AddressScheme (core: CRYPTO_ADDRESS_SCHEME_GEN_DEFAULT)
-        }
+        return defaultAddressSchemeMap[network.uids] ?? .genDefault
     }
 
     ///
+    /// Wallet Manager Modes
```
Comment (michael-brd) [inline Swift/BRCrypto/BRCryptoSystem.swift]: This should really be at the C level, but time is a factor.

File: Swift/BRCrypto/BRCryptoSystem.swift [hunk 4]
```diff
@@ -38,6 +38,118 @@ public final class System {
...
+         address: nil, verified: true,
+         demoninations: [(name: "drop", code: "drop", decimals: 0, symbol: "drop"),
+                         (name: "xrp",  code: "xrp",  decimals: 6, symbol: "xrp")]),
+    ]
+
+    ///
+    /// Address Scheme
+    ///
+
+    var supportedAddressSchemesMap: [String:[AddressScheme]] = [
```
Comment (michael-brd) [inline Swift/BRCrypto/BRCryptoSystem.swift]: This should really be at the C level, but time is a factor.

File: Swift/BRCrypto/BRCryptoSystem.swift (hunk 2 above)
Comment (EBGToo) [reply Swift/BRCrypto/BRCryptoSystem.swift]: We only add 'api_only' when BDB returns block height other than '-1' (BDB supports the blockchain).

File: Swift/BRCrypto/BRCryptoSystem.swift (hunk 3 above)
Comment (EBGToo) [reply Swift/BRCrypto/BRCryptoSystem.swift]: agreed.

File: Swift/BRCrypto/BRCryptoSystem.swift (hunk 4 above)
Comment (EBGToo) [reply Swift/BRCrypto/BRCryptoSystem.swift]: agreed.

File: Java/CoreCrypto/src/main/java/com/breadwallet/corecrypto/System.java [hunk 5]
```diff
@@ -235,13 +216,15 @@
     static {
         ImmutableMultimap.Builder<String, WalletManagerMode> builder = new ImmutableMultimap.Builder<>();
         builder.put("bitcoin-mainnet", WalletManagerMode.P2P_ONLY);
-        builder.put("bitcoin-cash-mainnet", WalletManagerMode.P2P_ONLY);
+        builder.put("bitcoin-mainnet", WalletManagerMode.API_ONLY);
+        builder.put("bitcoincash-mainnet", WalletManagerMode.P2P_ONLY);
```
[the remark this answers is not in the document]
Comment (michael-brd) [reply Java/CoreCrypto/src/main/java/com/breadwallet/corecrypto/System.java]: Will address in the CORE-515 branch.

**Pull request merged**